aiida/common/setup.py

```diff
@@ -638,6 +638,10 @@ def create_config_noninteractive(profile='default', force_overwrite=False, dry_r
     # finalizing
     write = not dry_run
     new_profile = update_profile(profile, new_profile, write=write)
+    old_profiles = [p for p in get_profiles_list() if p != profile]
```

Comment:
Isn't it better to switch this line above the previous one, and remove ``if p != profile``?
I'm not sure there is any guarantee that update_profile does not change the profile in a way that comparison becomes false (maybe if it adds some additional data, changes some capitalisation, etc.).